src/analyzers/ngram/ngram_word_analyzer.cpp

```diff
@@ -16,52 +16,61 @@ namespace meta
 namespace analyzers
 {
 
-const std::string ngram_word_analyzer::id = "ngram-word";
+template <class T>
+const std::string ngram_word_analyzer<T>::id = "ngram-word";
```

Comment:
We can probably create just `ngram_analyzer` and have it use different input streams, for example unigram words or POS tags. Not for this PR, but in general.